site/content/docs/contributing/1.0-roadmap.md

```diff
@@ -15,25 +15,41 @@ as a project.
 
 High level goals for kind include:
 
-- [ ] Reach [1.0] status / 
-stability by the end of Q1 2019 / the Kubernetes 1.14 release cycle
+## Beta Requirements
+
+To reach "beta" [grade][deprecation-policy] kind needs to at minimum:
+
+- [x] Improve documentation (Though this will eternally be "In Progress" !)
+  - [X] create a documentation site - [#268]
+  - [ ] expand examples of using kind
+  - [x] cover known issues, debugging, work-arounds, etc.
+- [x] Reliably [pass][kind-conformance-dashboard] the Kubernetes [conformance tests]
+ - [x] [Certify][certified] Conformance
 - [x] Support multi-node clusters - [#117]
-- [ ] Support offline / air-gapped clusters
-  - [ ] pre-loaded / offline CNI - [#200]
-- [ ] Support mounting host directories - [#62]
-- [ ] Improve Windows support
+- [x] Support offline / air-gapped clusters
+  - [x] pre-loaded / offline CNI - [#200]
+- [x] Support mounting host directories - [#62]
+- [x] Improve Windows support
   - [x] add Windows binaries to releases - [#155]
-  - [ ] improve instructions for KUBECONFIG in particular
-- [ ] Improve documentation - In Progress ...
-  - [ ] expand examples of using kind
-  - [ ] cover known issues, debugging, work-arounds, etc.
-  - [X] create a documentation site - [#268]
-- [ ] Improving logging and debuggability
-  - [ ] enable audit-logging
-- [ ] Support non-amd64 architectures (namely ARM) - [#166]
+  - [x] improve instructions for KUBECONFIG in particular
+- [ ] Support usage as a properly versioned, supported, and documented library 
+  - [ ] Improving logging and debuggability 
+for managing clusters in test harnesses
+- [ ] move API types / labels from `*.k8s.io` into [`*.x-k8s.io`][api-review-voluntary]
```

Comment:
@BenTheElder 
have you experimented with converters related to this change? TMK api-machinery does not support converting between types in a different Group. one way to solve this might be to keep types registered under both Groups, until the old API (`*.k8s.io) is deprecated.

the cluster API will do this (for alpha2 or 3) too but they use conversion web hooks, so they can tolerate such a change. given the challenges i see, i don't think we want to do that for projects like kops and kubeadm that have more mature APIs and user base.

i know that CertManager also uses `k8.io` (cc @munnerz) and SIG arch were giving it as an example of an external project that uses the "reserved" k8s.io domain.

Reply:
For kops and kubeadm I'm not sure, for kind we're early enough that it can even just be a breaking change if it comes down to it.

We have a lot of options given that these are not CRDs and the conversion only happens internally to Create(), shouldn't be a real problem.

I'm also increasingly convinced that for our use case the API machinery is more trouble than it's worth, we could implement conversion etc without it easily and already did once in a previous version of the code. Would probably keep deepcopy and skip the rest.

For the docker labels we will temporarily handle both but mark new things with the new one.

Reply:
> I'm also increasingly convinced that for our use case the API machinery is more trouble than it's worth, we could implement conversion etc without it easily and already did once in a previous version of the cod

the folks at WG Component Standard are also partially convinced at this point that API machinery is not really well suited for component configs WRT to validation and conversation.

yet the the WG is also driven by SIG api-machinery, so not sure how things would evolve there.

Reply:
I read this as migration from `k8s.io` to `x-k8s.io` for kind's API types only, nothing about attempting to convert or support migration of anyone else's types.  Given the intended lifetime of kind clusters, it's not clear to me how much time we want to invest in automated migration vs. guides or tooling for one-time changes.

I agree we should do this while in alpha when we're allowed to make clean breaks like this, and think beta is the appropriate boundary to stop using `k8s.io`

Reply:
er right, a clean break may have to be what we do.

I think we're all in agreement that we need to make this move, and it should happen for beta, @neolit123 was bringing up some technical aspects of how this might be done, and how this is a problem for many projects now.

I think we have a number of options to achieve this with greater ease in kind, up to and including just dropping support for the old API group if we have to. That's a discussion best had elsewhere though :sweat_smile: